app/commands/payments/payment/update_amount.rb

```diff
@@ -8,6 +8,7 @@ def call
     return unless payment
 
     payment.update!(amount_in_cents:)
-    User::UpdateTotalDonatedInCents.(payment.user)
+    User::UpdateTotalDonatedInCents.(payment.user) if payment.donation?
+    User::Premium::Update.(user) if payment.premium?
```

Comment:
```suggestion
    if payment.donation?
      User::UpdateTotalDonatedInCents.(payment.user) 
    elsif payment.premium?
      User::Premium::Update.(user) 
    end
```

I just find this a bit clearer.